lib/routes/charge/pos_invoice_items_view.dart

```diff
@@ -32,62 +34,54 @@ class PosInvoiceItemsView extends StatelessWidget {
     return StreamBuilder(
       stream: posCatalogBloc.itemsStream,
       builder: (context, snapshot) {
-        var posCatalog = snapshot.data;
+        final posCatalog = snapshot.data;
         if (posCatalog == null) return Loader();
 
-        return Scaffold(
-          body: _buildCatalogContent(
-            context,
-            posCatalog,
-          ),
-          floatingActionButton: FloatingActionButton(
-            child: Icon(
-              Icons.add,
-            ),
-            backgroundColor: themeData.buttonColor,
-            foregroundColor: themeData.textTheme.button.color,
-            onPressed: () => Navigator.of(context).pushNamed("/add_item"),
-          ),
+        return StreamBuilder(
+          stream: posCatalogBloc.posItemSort,
+          initialData: PosCatalogItemSort.NONE,
+          builder: (context, snapshot) {
+            final sort = snapshot.data ?? PosCatalogItemSort.NONE;
+
+            return Scaffold(
+              body: _buildCatalogContent(
+                context,
+                posCatalogBloc,
+                _sort(posCatalog, sort),
```

Comment:
I think it is better to sort in the bloc and leave the UI as is. This will allow more options in the future such as paging, virtual list, etc...
What do you think about sorting the list in itemsStreamController whenever sorting changes?

Reply:
Excelente suggestion, now it is sorted on the bloc :)